### DISCUSSION TITLE
"knitr buttons missing" in targets chunks

### DISCUSSION THREAD

**Discussion by christianholland** [ropensci/targets#707] [Help]
## Prework

* [x] Read and agree to the [code of conduct](https://ropensci.org/code-of-conduct/) and [contributing guidelines](https://github.com/ropensci/targets/blob/main/CONTRIBUTING.md).
* [x] Confirm that your issue is a genuine bug in the `targets` package itself and not a user error, known limitation, or issue from another package that `targets` depends on. For example, if you get errors running `tar_make_clustermq()`, try isolating the problem in a reproducible example that runs `clustermq` and not `targets`. And for miscellaneous troubleshooting, please post to [discussions](https://github.com/ropensci/targets/discussions) instead of [issues](https://github.com/ropensci/targets/issues).
* [x] If there is [already a relevant issue](https://github.com/ropensci/targets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
## Description

Hi there,

When using the targets engine in an Rmarkdown file I am missing the buttons (`Modify Chunk Options`, `Run All Chunks Above`, `Run Current Chunk`) in the top right corner of the individual chunk (see attached screenshot). 

![image](https://user-images.githubusercontent.com/28303875/142628579-9d582468-3d91-49ee-bdd2-981669ff382e.png)

This leads to the problem whenI am executing the chunk in an interactive session with `Ctrl`+`Enter` it doesnt write the objects into the environment but returns a list of `tar_stem` (?) (see below)

```
[[1]]
<tar_stem> 
  name: sample_anno 
  command:
    read_sample_annotation(sample_annotation_file) 
  format: rds 
  iteration method: vector 
  error mode: stop 
  memory mode: persistent 
  storage mode: main 
  retrieval mode: main 
  deployment mode: worker 
  priority: 0 
  resources:
    list() 
  cue:
    mode: thorough
    command: TRUE
    depend: TRUE
    format: TRUE
    iteration: TRUE
    file: TRUE 
  packages:
    targets
    stats
    graphics
    grDevices
    datasets
    utils
    methods
    base 
  library:
    NULL
[[2]]
<tar_stem> 
  name: sample_mapping 
  command:
    read_sample_mapping(sample_mapping_file) 
  format: rds 
  iteration method: vector 
  error mode: stop 
  memory mode: persistent 
  storage mode: main 
  retrieval mode: main 
  deployment mode: worker 
  priority: 0 
  resources:
    list() 
  cue:
    mode: thorough
    command: TRUE
    depend: TRUE
    format: TRUE
    iteration: TRUE
    file: TRUE 
  packages:
    targets
    stats
    graphics
    grDevices
    datasets
    utils
    methods
    base 
  library:
    NULL
```

**Comment by wlandau** [comment]
> When using the targets engine in an Rmarkdown file I am missing the buttons (Modify Chunk Options, Run All Chunks Above, Run Current Chunk) in the top right corner of the individual chunk (see attached screenshot).

Unfortunately, `targets` does not control which buttons appear in the [notebook interface](https://bookdown.org/yihui/rmarkdown/notebook.html). The [RStudio IDE](https://github.com/rstudio/rstudio) controls that part, and it could have something to do with your particular installation. 

> This leads to the problem when I am executing the chunk in an interactive session with Ctrl+Enter it doesnt write the objects into the environment but returns a list of tar_stem (?) (see below)

Target Markdown uses a custom [knitr language engine](https://bookdown.org/yihui/rmarkdown-cookbook/custom-engine.html) which is registered when either `targets` >= 0.5.0 or `knitr` >= 1.34 is loaded. If the package versions are up to date and `library(targets)` does not allow interactive mode to work, then it could be another issue with the IDE trying to use the R language engine instead of the `targets` language engine.
